File: backend/src/models/transaction.py
from pydantic import BaseModel
from datetime import datetime

class Transaction(BaseModel):
  text: str
  date: datetime
  inputSize: int

  sha256Time: float
  sha256OutputSize: int
  sha256ArbitrumPrice: float
  sha256OptimismPrice: float
  sha256BasePrice: float

  keccak256Time: float
  keccak256OutputSize: int
  keccak256ArbitrumPrice: float
  keccak256OptimismPrice: float
  keccak256BasePrice: float

  aes256Time: float
  aes256OutputSize: int
  aes256ArbitrumPrice: float
  aes256OptimismPrice: float
  aes256BasePrice: float

  rsaTime: float
  rsaOutputSize: int
  rsaArbitrumPrice: float
  rsaOptimismPrice: float
  rsaBasePrice: float

File: vueProject/frontend/src/types/Transaction.ts
export interface Transaction {
  text: string
  date: string
  inputSize: number

  sha256Time: number
  sha256OutputSize: number
  sha256ArbitrumPrice: number
  sha256OptimismPrice: number
  sha256BasePrice: number

  keccak256Time: number
  keccak256OutputSize: number
  keccak256ArbitrumPrice: number
  keccak256OptimismPrice: number
  keccak256BasePrice: number

  aes256Time: number
  aes256OutputSize: number
  aes256ArbitrumPrice: number
  aes256OptimismPrice: number
  aes256BasePrice: number

  rsaTime: number
  rsaOutputSize: number
  rsaArbitrumPrice: number
  rsaOptimismPrice: number
  rsaBasePrice: number
}
